python/ray/data/datasource/file_based_datasource.py

```diff
@@ -217,6 +220,13 @@ def expand_paths(
     ) -> Tuple[List[str], List[Optional[int]]]:
         from pyarrow.fs import FileType
 
+        if len(paths) > 1:
```

Comment:
Why is this a good indicator to log? The user can supply a mix of files and directories in the 'paths', right?

Reply:
The assumption here is that the most common use-case is that (1) a single directory is passed containing input files or (2) multiple paths are passed containing input files. Mixed file and directory inputs are assumed to be less common. However, I think I could also change this to only log the warning after a fixed amount of time has elapsed expanding directories, which may be more generally appropriate.

Reply:
Ok, it seems fine as it's simpler (since it's just a precaution).